code/range/ViewModel/src/main/java/ru/unn/agile/range/viewmodel/ViewModel.java

```diff
@@ -0,0 +1,186 @@
+package ru.unn.agile.range.viewmodel;
+
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.property.StringProperty;
+import ru.unn.agile.range.model.Range;
+
+import java.util.Arrays;
+
+public class ViewModel {
+
+    private BooleanProperty btnGetEndPointsDisabled = new SimpleBooleanProperty();
+    private BooleanProperty btnGetAllPointsDisabled = new SimpleBooleanProperty();
+    private BooleanProperty btnContainsDisabled = new SimpleBooleanProperty();
+    private BooleanProperty btnEqualsDisabled = new SimpleBooleanProperty();
+    private BooleanProperty btnOverlapsDisabled = new SimpleBooleanProperty();
+    private StringProperty txtInput = new SimpleStringProperty();
+    private StringProperty txtRange = new SimpleStringProperty();
+    private StringProperty txtResult = new SimpleStringProperty();
+
+    private Range range;
+
+    public ViewModel() {
+        btnGetEndPointsDisabled.setValue(true);
+        btnGetAllPointsDisabled.setValue(true);
+        btnOverlapsDisabled.setValue(true);
+        btnEqualsDisabled.setValue(true);
+        btnContainsDisabled.setValue(true);
+
+        txtResult.setValue("");
+        txtRange.setValue("");
+        txtInput.setValue("");
+
+        txtRange.addListener((observable, oldValue, newValue) -> {
+            setRange(newValue);
+        });
+        txtInput.addListener((observable, oldValue, newValue) -> {
+            setInput(newValue);
+        });
+    }
+
+    public BooleanProperty isGetEndPointsButtonDisabled() {
+        return btnGetEndPointsDisabled;
+    }
+
+    public BooleanProperty isGetAllPointsButtonDisabled() {
+        return btnGetAllPointsDisabled;
+    }
+
+    public BooleanProperty isContainsButtonDisabled() {
+        return btnContainsDisabled;
+    }
+
+    public BooleanProperty isEqualsButtonDisabled() {
+        return btnEqualsDisabled;
+    }
+
+    public BooleanProperty isOverlapsButtonDisabled() {
+        return btnOverlapsDisabled;
+    }
+
+    public StringProperty getTxtRange() {
+        return txtRange;
+    }
+
+    public StringProperty getTxtResult() {
+        return txtResult;
+    }
+
+    public StringProperty getTxtInput() {
+        return txtInput;
+    }
+
+    public void containsInput() {
+        String input = txtInput.get();
+
+        if (isInteger(input)) {
+            if (range.containsValue(Integer.parseInt(input))) {
+                txtResult.setValue("Yes");
+            } else {
+                txtResult.setValue("No");
+            }
+        }
+        if (isIntegerSet(input)) {
+            if (range.containsSet(Arrays.stream(input.substring(1, input.length() - 1).split(","))
```

Comment:
Instead split(",") you can use split("[,\\s]+") for cases when space between numbers

Reply:
It was done intentionally. I don't see any reason to allow range with spaces inside:)